python/ray/serve/controller.py

```diff
@@ -425,6 +426,17 @@ class Checkpoint:
     # TODO(ilr) Rename reconciler to PendingState
 
 
+@dataclass
+class FutureResult:
+    # Whether API call is successful
+    success: bool
+    # Goal requested when this future was created
+    requested_goal: Dict[str, Any]
+    # Goal state that supercedes (cancels) the requested goal
+    successor_goal: Optional[Dict[str, Any]] = field(
+        default_factory=lambda: None)
```

Comment:
```suggestion
    successor_goal: Optional[Dict[str, Any]] = None
```
Just setting the default should be fine. I think `field(default_factory=)` pattern is only used for mutable defaults.

Reply:
Ahhh thanks!